### PR TITLE
20_2 TreeList Form Editing: rewrite

### PR DESCRIPTION
**Pull request by pavel-2020**
[Trello](https://trello.com/c/gGAbEsoc/1650-treelist-demos-editing-5-demos)
https://github.com/DevExpress/devextreme-demos/pull/826 https://github.com/DevExpress/devextreme-demos/pull/827

### CODE REVIEW

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 1]
```diff
@@ -1 +1,19 @@
...
+- **editing**.[allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating) to **true**
+
+A column's default editor is defined automatically based on this column's [dataType](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#dataType). To customize this editor or replace it with another editor, use the column's [formItem](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#formItem) object. Refer to the [SimpleItem](/Documentation/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/) help topic for information on settings that you can define in the **formItem** object.
+
+You can also set up the edit Form from scratch. The component's [configuration section](/Documentation/ApiReference/UI_Components/dxForm/) lists available settings. Specify these settings in the **editing**.[form](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#form) object. Refer to this object's description for more information about this edit mode's limitations.
+
+This demo uses form editing mode and illustrates the following event handlers:
```
Comment (AlexJameson) [inline JSDemos/Demos/TreeList/FormEditing/description.md]: I think its is quite obvious:

```suggestion
This demo also illustrates the following event handlers:
```

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 2]
```diff
@@ -1 +1,19 @@
...
+
+- [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow)        
+Populates form editors of a new row with default values. In this demo, **onInitNewRow** sets John Heart as the initial Head for new rows.
```
Comment (AlexJameson) [inline JSDemos/Demos/TreeList/FormEditing/description.md]:
```suggestion
Populates form editors of a new row with default values. In this demo, **onInitNewRow** sets *"John Heart"* as the initial `Head` for new rows.
```

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 3]
```diff
@@ -1 +1,19 @@
...
+
+- [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing)         
+Customizes form editors. In this demo, this handler disallows users to edit the Head value of John Heart.
```
Comment (AlexJameson) [inline JSDemos/Demos/TreeList/FormEditing/description.md]:
```suggestion
Customizes form editors. In this demo, this handler disallows users to edit the `Head` value of John Heart.
```

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 4]
```diff
@@ -1 +1,19 @@
...
\ No newline at end of file
+TreeList can use the [Form](/Documentation/ApiReference/UI_Components/dxForm/) component to arrange cell editors when users modify a row. Users can edit values in [visible](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visible) and hidden columns.
+
+To use the Form as the row editor UI, specify the following properties:
+- **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) to *"form"*
+- **editing**.[allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating) to **true**
+
+A column's default editor is defined automatically based on this column's [dataType](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#dataType). To customize this editor or replace it with another editor, use the column's [formItem](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#formItem) object. Refer to the [SimpleItem](/Documentation/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/) help topic for information on settings that you can define in the **formItem** object.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/TreeList/FormEditing/description.md]: Two sentences in a row end with the same phrase.

```suggestion
A column's default editor is defined automatically based on this column's [dataType](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#dataType). To customize this editor or replace it with another editor, use the column's [formItem](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#formItem) object. For information on settings that you can define in the **formItem** object, refer to the [SimpleItem](/Documentation/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/) help topic.
```

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 5]
```diff
@@ -1 +1,19 @@
...
+
+You can also set up the edit Form from scratch. The component's [configuration section](/Documentation/ApiReference/UI_Components/dxForm/) lists available settings. Specify these settings in the **editing**.[form](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#form) object. Refer to this object's description for more information about this edit mode's limitations.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/TreeList/FormEditing/description.md]: No information about the limitations has been given yet.

```suggestion
You can also set up the edit Form from scratch. The component's [configuration section](/Documentation/ApiReference/UI_Components/dxForm/) lists available settings. Specify these settings in the **editing**.[form](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#form) object. Refer to this object's description for information about form edit mode's limitations.
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — DISMISSED** [review]

**Review by AlexJameson — DISMISSED** [review]

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 6]
```diff
@@ -1 +1,19 @@
...
+- **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) to *"form"*
+- **editing**.[allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating) to **true**
+
+A column's default editor is defined automatically based on this column's [dataType](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#dataType). To customize this editor or replace it with another editor, use the column's [formItem](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#formItem) object. For information on settings that you can define in the **formItem** object, refer to the [SimpleItem](/Documentation/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/) help topic.
+
+You can also set up the edit Form from scratch. The component's [configuration section](/Documentation/ApiReference/UI_Components/dxForm/) lists available settings. Specify these settings in the **editing**.[form](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#form) object. Refer to this object's description for information about form edit mode's limitations.
```
Comment (dmitry-eliseev-devexpress) [inline JSDemos/Demos/TreeList/FormEditing/description.md]:
```suggestion
You can also set up the edit Form from scratch. The component's [configuration section](/Documentation/ApiReference/UI_Components/dxForm/) lists available settings. Specify these settings in the **editing**.[form](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#form) object. Refer to this object's description for information about form edit mode limitations.
```

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 7]
```diff
@@ -1 +1,19 @@
...
+
+This demo also illustrates the following event handlers:
+
+- [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow)        
+Populates form editors of a new row with default values. In this demo, **onInitNewRow** sets *"John Heart"* as the initial `Head` for new rows.
+
+- [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing)         
+Customizes form editors. In this demo, this handler disallows users to edit the `Head` value of John Heart.
```
Comment (dmitry-eliseev-devexpress) [inline JSDemos/Demos/TreeList/FormEditing/description.md]: **dis**allows is a negative statement. It's better to choose a word that does not require negation to convey the same idea.

```suggestion
Customizes form editors. In this demo, this handler forbids users to edit the `Head` value of John Heart.
```

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 8]
```diff
@@ -1 +1,19 @@
-This example demonstrates editing in _&laquo;form&raquo;_ mode. To&nbsp;begin to&nbsp;edit a&nbsp;row, click its _&laquo;Edit&raquo;_ link. On&nbsp;switching into the editing state, this row will become a&nbsp;form with editable fields.
\ No newline at end of file
+TreeList can use the [Form](/Documentation/ApiReference/UI_Components/dxForm/) component to arrange cell editors when users modify a row. Users can edit values in [visible](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visible) and hidden columns.
```
Comment (dmitry-eliseev-devexpress) [inline JSDemos/Demos/TreeList/FormEditing/description.md]: From what I understand, the editors on the Form allow users to modify values both in visible and hidden columns. However, the second sentence here does not convey this idea clearly enough. Right now to me it sounds like users always can edit all kinds of columns. Please take another look at this paragraph. You most likely can come up with something better.

File: JSDemos/Demos/TreeList/FormEditing/description.md [hunk 9]
```diff
@@ -1 +1,19 @@
-This example demonstrates editing in _&laquo;form&raquo;_ mode. To&nbsp;begin to&nbsp;edit a&nbsp;row, click its _&laquo;Edit&raquo;_ link. On&nbsp;switching into the editing state, this row will become a&nbsp;form with editable fields.
\ No newline at end of file
+The TreeList can use the [Form](/Documentation/ApiReference/UI_Components/dxForm/) component to arrange cell editors when users modify a row. The Form allows users to edit values from both [visible and hidden](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visible) columns.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/TreeList/FormEditing/description.md]:
```suggestion
The TreeList can use the [Form](/Documentation/ApiReference/UI_Components/dxForm/) component to arrange cell editors when users modify a row. The Form allows users to edit values from [visible and hidden](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visible) columns.
```

File: JSDemos/Demos/DataGrid/FormEditing/description.md [hunk 10]
```diff
@@ -1,4 +1,4 @@
-DataGrid can use the [Form](/Documentation/ApiReference/UI_Components/dxForm/) component to arrange cell editors when users modify a row. Users can edit value in [visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#visible) and hidden columns (see the `Notes` column).
+The DataGrid can use the [Form](/Documentation/ApiReference/UI_Components/dxForm/) component to arrange cell editors when users modify a row. The Form allows users to edit values from both [visible and hidden](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#visible) columns (see the `Notes` column).
```
Comment (dirk-pieterse) [inline JSDemos/Demos/DataGrid/FormEditing/description.md]:
```suggestion
The DataGrid can use the [Form](/Documentation/ApiReference/UI_Components/dxForm/) component to arrange cell editors when users modify a row. The Form allows users to edit values from [visible and hidden](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#visible) columns (see the `Notes` column).
```

**Review by dirk-pieterse — DISMISSED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**